src/v2/BaseLoginRouter.js

```diff
@@ -245,6 +249,32 @@ export default Router.extend({
     this.controller.render();
   },
 
+  /**
+    * When "Remember My Username" is enabled, we save the identifier in a cookie
+    * so that the next time the user visits the SIW, the identifier field can be 
+    * pre-filled with this value.
+   */
+  updateIdentifierCookie: function(idxResponse) {
+    // Only update the cookie when the user has successfully authenticated themselves 
+    // to avoid incorrect/uneccessary updates. This is done by checking the server response
+    // and seeing if either the 'success' or 'successWithInteractionCode' objects are present.
+    if (!idxResponse?.rawIdxState?.success && !idxResponse?.rawIdxState?.successWithInteractionCode) {
```

Comment:
just for the sake of readability create another function

`isAuthenticationSucceeded()`

Reply:
I'll add this in.